cmd/skopeo/sync.go

```diff
@@ -380,6 +384,68 @@ func imagesToCopyFromRegistry(registryName string, cfg registrySyncConfig, sourc
 			Context:      serverCtx})
 	}
 
+	for imageName, semVerCompare := range cfg.ImagesBySemVerCompare {
```

Comment:
This function is by now way too long and redundant.

Probably the `len(tags) != 0` case should be handled specially; the other three (`len(tags) == 0`, `ImagesByTagRegex`, `ImagesBySemVerCompare` should probably use a common helper that does the query+loops, parameterized by a “filter” callback.

And the it would be useful to have a 5-10-item unit test for the semantic versioning filter.

Reply:
I'm not really a golong person but gave it a shot. I've taken out the regex and semver filtering. The plain tags/empty list one is a bit too different to easily go in  the same function.

I've also added some unit-tests for filterRegex and filterSemVer.